File: group#2/김명기/1주차/2.py
# 행렬 테두리 회전하기

def solution(rows, columns, queries):
    answer = []
    arr = [[0] * columns for i in range(rows)]
    for i in range(rows * columns):
        arr[i//columns][i % columns] = i+1
    for query in queries:
        x1, y1, x2, y2 = query
        x1 -= 1
        y1 -= 1
        x2 -= 1
        y2 -= 1
        brr = [[0] * columns for i in range(rows)]
        m = 987654321
        x = x1
        for y in range(y1, y2):
            brr[x][y+1] = arr[x][y]
            m = min(m, arr[x][y])
        y = y2
        for x in range(x1, x2):
            brr[x+1][y] = arr[x][y]
            m = min(m, arr[x][y])
        x = x2
        for y in range(y2, y1, -1):
            brr[x][y-1] = arr[x][y]
            m = min(m, arr[x][y])
        y = y1
        for x in range(x2, x1, -1):
            brr[x-1][y] = arr[x][y]
            m = min(m, arr[x][y])
        for i in range(rows):
            for j in range(columns):
                if brr[i][j] != 0:
                    arr[i][j] = brr[i][j]
        answer.append(m)
    return answer

File: group#2/김수현/1주차/2.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int r, int c, vector<vector<int>> q) {
    vector<vector<int>> a(r,vector<int>(c));
    for(int i=0;i<r;i++){
        for(int j=0;j<c;j++){
            a[i][j]=i*c+1+j;
        }
    }
    vector<int> answer;
    for(int i=0;i<q.size();i++){
        int x1=q[i][0]-1,x2=q[i][2]-1,y1=q[i][1]-1,y2=q[i][3]-1;
        int minnum = 10000;
        int cur = a[x1][y1];
        for(int j=y1+1;j<=y2;j++){
            minnum = min(minnum,cur);
            int temp = a[x1][j];
            a[x1][j]=cur;
            cur = temp;
        }
        for(int j=x1+1;j<=x2;j++){
            minnum = min(minnum,cur);
            int temp = a[j][y2];
            a[j][y2]=cur;
            cur = temp;
        }
        for(int j=y2-1;j>=y1;j--){
            minnum = min(minnum,cur);
            int temp = a[x2][j];
            a[x2][j]=cur;
            cur = temp;
        }
        for(int j=x2-1;j>=x1;j--){
            minnum = min(minnum,cur);
            int temp = a[j][y1];
            a[j][y1]=cur;
            cur=temp;
        }
        minnum = min(minnum,cur);
        answer.push_back(minnum);
    }
    
    return answer;
}
